nixos/modules/services/matrix/mautrix-meta.nix

```diff
@@ -0,0 +1,489 @@
+            fi
+
+            # Generation overrides as_token, hs_token.
+            if [[ "$generated" == "1" || "$has_tokens" == "false" ]]; then
+              echo "Substituting as_token, hs_token inside the config file"
+              ${lib.getExe pkgs.yq} -s '.[0].appservice.as_token = .[1].as_token
+                | .[0].appservice.hs_token = .[1].hs_token
+                | .[0]' '${settingsFile cfg}' '${cfg.registrationFile}' \
+                > '${settingsFile cfg}.tmp'
+              mv '${settingsFile cfg}.tmp' '${settingsFile cfg}'
+            fi
+
+            umask $old_umask
+          ''; #+ lib.optionalString config.services.matrix-synapse.enable ''
+          #   chown mautrix-meta:matrix-synapse '${cfg.registrationFile}'
+          # '';
+
+          serviceConfig = {
+            UMask = 0027;
+
+            User = "mautrix-meta";
+            Group = "mautrix-meta";
+
+            SystemCallFilter = [ "@system-service" "@chown" ];
+            CapabilityBoundingSet = [ "CAP_CHOWN" ];
+
+            ProtectSystem = "strict";
+            ProtectHome = true;
+
+            StateDirectory = cfg.dataDir;
+            ReadWritePaths = fullDataDir cfg;
+          };
+
+          restartTriggers = [ (settingsFileUnsubstituted cfg) ];
+        };
+          "${metaName name}" = {
+            description = "Mautrix-Meta bridge - ${metaName name}";
+            wantedBy = [ "multi-user.target" ];
+            wants = [ "network-online.target" ] ++ cfg.serviceDependencies;
+            after = [ "network-online.target" ] ++ cfg.serviceDependencies;
+
+            environment.HOME = fullDataDir cfg;
+
+            serviceConfig = {
+              Type = "simple";
+
+              User = "mautrix-meta";
+              Group = "mautrix-meta";
+              PrivateUsers = true;
+
+              LockPersonality = true;
+              MemoryDenyWriteExecute = true;
+              NoNewPrivileges = true;
+              PrivateDevices = true;
+              PrivateTmp = true;
+              ProtectClock = true;
+              ProtectControlGroups = true;
+              ProtectHome = true;
+              ProtectHostname = true;
+              ProtectKernelLogs = true;
+              ProtectKernelModules = true;
+              ProtectKernelTunables = true;
+              ProtectSystem = "strict";
+              Restart = "on-failure";
+              RestartSec = "30s";
+              RestrictRealtime = true;
+              RestrictSUIDSGID = true;
+              SystemCallArchitectures = "native";
+              SystemCallErrorNumber = "EPERM";
+              SystemCallFilter = ["@system-service"];
+              UMask = 0027;
+
+              WorkingDirectory = fullDataDir cfg;
+              ReadWritePaths = fullDataDir cfg;
+              StateDirectory = cfg.dataDir;
+              EnvironmentFile = cfg.environmentFile;
+
+              ExecStart = "${lib.getExe upperCfg.package} --config='${settingsFile cfg}'";
```

Comment:
```suggestion
              ExecStart = lib.escapeShellArgs [
                (lib.getExe upperCfg.package)
                "--config=${settingsFile cfg}"
		      ];
```

Reply:
That's better, changed it to this.

Reply:
Sorry, what is the issue now?

Reply:
Don't worry about it. Basically if the user configures data dir or registration file with a `'` or something we will have a bad time. But this is basically true of every nixos module...

Reply:
Oh, right!